python/pyspark/sql/types.py

```diff
@@ -368,8 +367,49 @@ def __init__(self, fields):
         >>> struct1 == struct2
         False
         """
-        assert all(isinstance(f, DataType) for f in fields), "fields should be a list of DataType"
-        self.fields = fields
+        if not fields:
+            self.fields = []
+        else:
+            self.fields = fields
+            assert all(isinstance(f, StructField) for f in fields),\
+                "fields should be a list of StructField"
+
+    def add(self, name_or_struct_field, data_type=NullType(), nullable=True, metadata=None):
+        """
+        Construct a StructType by adding new elements to it to define the schema
+        >>> struct1 = StructType().add("f1", StringType(), True).add("f2", StringType(), True, None)
+        >>> struct2 = StructType([StructField("f1", StringType(), True),\
```

Comment:
The doctests will be taken as examples in API docs, should be simple. We could put other test cases into sql/tests.py